graph-deploy.js

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env node
+
+let app = require('commander')
+let request = require('request')
+let args = require('./src/cli/args')
+const url = require('url')
+
+args.addBuildCommand()
+
+app
+    .option(
+        '-n, --subgraph-name <NAME>',
+        'subgraph name'
+    )
+    .option(
+        '--node <URL>[:PORT]',
+        'graph node url'
+    )
+
+app.parse(process.argv)
+
+if (!app.node || !app.subgraphName) {
+    app.help()
+}
+
+let compiler = args.compilerFromArgs()
+
+let requestUrl = new URL(app.node);
+if (!requestUrl.port) {
+    requestUrl.port = "8020"
+}
+
+compiler.compile()
+    .then(
+        function (ipfsHash) {
+            if (ipfsHash === undefined) {
+                console.log("compilation failed, not deploying")
```

Comment:
`compile` already logs something, so here we could just exit with an exit code of 1:
```js
process.exitCode = 1
return
```
There is also `process.exit(1)` but that [is not recommended](https://stackoverflow.com/questions/5266152/how-to-exit-in-node-js/37592669#37592669).